conbench/entities/_entity.py

```diff
@@ -19,6 +19,18 @@ class NotFound(NoResultFound):
     pass
 
 
+class EntityExists(Exception):
+    """
+    Custom exception representing a database conflict. Meant to be caught from
+    within HTTP request handlers, and to be translated into an HTTP response
+    with status code 409. The `msg` in `raise EntityExists(msg)` is meant to be
+    exposed directly to the HTTP client (that is, construct it mindfully: do
+    not expose meaningless detail).
+    """
```

Comment:
Does this bubble up into the API docs? If no, should we (or did I miss it somewhere...)?

Reply:
This docstring here is developer documentation, for how to use `EntityExists` in the webapp's Python code.

We may soon want to add a generic statement to our HTTP API specification that for all POST requests for entity management a to-be-expected HTTP response is one with status code 409, typically indicating an insertion/creation conflict (but the exact nature of the error is clarified in the body of the response). We will do that. This is the same category as documenting a 404 response for "entity not found".